sdk/core/System.ClientModel/tests/TestFramework/Mocks/MockPipelineResponse.cs

```diff
@@ -73,4 +101,48 @@ protected void Dispose(bool disposing)
             _disposed = true;
         }
     }
+
+    public override BinaryData ReadContent(CancellationToken cancellationToken = default)
+    {
+        if (_bufferedContent is not null)
+        {
+            return _bufferedContent;
+        }
+
+        if (_contentStream is null)
+        {
+            _bufferedContent = BinaryData.FromString(string.Empty);
+            return _bufferedContent;
+        }
+
+        MemoryStream bufferStream = new();
+        _contentStream.CopyTo(bufferStream);
+        _contentStream.Dispose();
+        _contentStream = bufferStream;
+
+        _bufferedContent = BinaryData.FromStream(bufferStream);
```

Comment:
is FromStream optimized for MemoryStream? If not, this does unnecessary copy/stream-read.

Reply:
Since this is a mock, I'll leave this, but I'll add a comment to refer anyone wishing to copy it to a more performant implementation.